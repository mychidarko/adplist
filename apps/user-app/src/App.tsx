import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import './App.css';
import Login from './pages/auth/login';
import Profile from './pages/profile';

function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Profile />} />
        <Route path="/auth/login" element={<Login />} />
      </Routes>
    </Router>
  );
}

export default App;
