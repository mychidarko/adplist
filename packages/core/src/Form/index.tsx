import * as React from "react";
import { hasError } from "./utils";
import { FormProps } from "./@types/Form";

export const Form: React.FC<FormProps> = ({ form, onSubmit, children, ...rest }) => {
	// we use this errors state to handle the input focusing in the layout effect
	// not sure if this duplication is neccessary but it works so we leave it for now
	const [errors, setErrors] = React.useState({});

	function _submit(e: any) {
		if (!form) return;
		e.preventDefault();

		const errors = form.validateFields();

		setErrors(errors);

		if (!hasError(errors)) {
			return onSubmit(e, form.inputState);
		}
	}

	React.useEffect(() => {
		const errorInputs = document.querySelectorAll(".input-error");
		const errorInput: any = errorInputs[0];

		if (errorInput) {
			errorInput.focus();
		}
	}, [errors]);

	return <form onSubmit={_submit} {...rest}>{children}</form>;
};

export { default as Button } from "./Button";
export { Select, CheckBox, Input, TextArea } from "./Input/Input";
export { default as useForm } from "./hooks/useForm";

