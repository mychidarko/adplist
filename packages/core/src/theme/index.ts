import colors from "./colors";
import gradients from "./gradients";
import presets from "./presets";
import typography from "./typography";
import shadows from "./shadows";
import animations from "./animations";
import breakpoints from "./breakpoints";

const theme = {
	colors,
	gradients,
	presets,
	typography,
	shadows,
	animations,
	breakpoints,
};

export default theme;
