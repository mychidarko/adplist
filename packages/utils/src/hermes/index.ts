import axios, { AxiosPromise, AxiosResponse } from "axios";
import { useEffect, useState } from "react";
import { HermesData, HookResponse } from "./@types/hermes";

export class HermesConfig {
	static baseUrl: string = "";
	static token: string;
	static parseErr: boolean = true;
	static blockOffline: boolean = true;
	static onAuthError: Function;
	static onServerError: Function;
	static withCredentials: boolean = false;
};

export function hermes({
	data,
	url,
	method = "GET",
	token = null,
  authorization = null,
	cache = null,
	withCredentials,
	useBase = true,
	options = {
		parseErr: HermesConfig.parseErr,
		blockOffline: HermesConfig.blockOffline,
	},
}: HermesData): AxiosPromise {
	const headers: any = {};
	const requestURL = useBase ? `${HermesConfig.baseUrl}${url}` : url;

	if (authorization) {
    headers.Authorization = authorization;
  } else {
    if (token) {
      headers.Authorization = `Bearer ${token}`;
    } else if (HermesConfig.token) {
      headers.Authorization = `Bearer ${HermesConfig.token}`;
    }
  }

	return new Promise(async (resolve, reject) => {
		if (!window.navigator.onLine && options.blockOffline) {
			return reject(
				"No network detected! This feature requires an internet connection!"
			);
		}

		try {
			const response = await axios({
				data,
				url: requestURL,
				method,
				headers,
				withCredentials: withCredentials || HermesConfig.withCredentials,
			});

			if (!response.data) {
				response.data = {
					data: cache,
					mode: "CACHE",
				};
			}

			return resolve(response);
      // @ts-ignore
		} catch (err: any) {
			if (!options.parseErr) {
				return reject(err);
			}

			if (err.response) {
				const onAuthError =
					options.onAuthError || HermesConfig.onAuthError;

				if (err.response.status === 401 && onAuthError) {
					onAuthError(err);
				}

				const onServerError =
					options.onServerError || HermesConfig.onServerError;

				if (err.response.status === 500 && onServerError) {
					onServerError(err);
				}

				return reject(err.response);
			}

			if (err.message) {
				return reject(err.message);
			}

			const errors = err.split(" ");
			let errorString = "";

			for (let index = 0; index < errors.length; index + 1) {
				const error = errors[index];

				if (index > 0) {
					errorString = `${errorString} ${error} `;
				}
			}

			return reject(errorString);
		}
	});
}

export function useHermes(
	url: string,
	options: Partial<HermesData> = {
		method: "GET",
		refresh: 0,
	}
): HookResponse {
	const [loading, setLoading] = useState<boolean>(
		options.cache !== null || true
	);
	const [response, setResponse] = useState<Partial<AxiosResponse<any>>>({});
	const [error, setError] = useState(null);

	const makeRequest = () => {
		hermes({ url, ...options })
			.then((res) => {
				if (!res && options.cache) {
					const response: any = res;

					response.data = {
						data: options,
						mode: "CACHE",
					};

					setResponse(response);
				}

				setResponse(res);
			})
			.catch((err) => {
				setError(err);
			})
			.finally(() => {
				setLoading(false);
			});
	};

	const runQuery = () => {
		makeRequest();

		if (options.refresh) {
			const refresh = setInterval(() => {
				makeRequest();
			}, options.refresh);

			return () => {
				clearInterval(refresh);
			};
		}

    return () => {};
	};
	// eslint-disable-next-line react-hooks/exhaustive-deps
	useEffect(runQuery, [url]);

	// eslint-disable-next-line react-hooks/exhaustive-deps
	useEffect(runQuery, []);

  // @ts-ignore
	return [response, loading, error];
}

export function useGet(url: string, options: Partial<HermesData> = {}) {
	return useHermes(url, {
		method: "GET",
		...options,
	});
}

export function withGet() {
	return useGet;
}

export function usePost(url: string, options: Partial<HermesData> = {}) {
	return useHermes(url, {
		method: "POST",
		...options,
	});
}

export function usePut(url: string, options: Partial<HermesData> = {}) {
	return useHermes(url, {
		method: "PUT",
		...options,
	});
}
