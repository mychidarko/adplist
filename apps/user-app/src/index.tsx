import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
import 'bootstrap/dist/css/bootstrap.css';
import App from './App';
import reportWebVitals from './reportWebVitals';
import { Auth0Provider } from '@auth0/auth0-react';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <Auth0Provider
      domain={process.env.REACT_APP_AUTH0_DOMAIN!}
      clientId={process.env.REACT_APP_AUTH0_CLIENT_ID!}
      redirectUri={
        typeof window !== 'undefined'
          ? window.location.origin
          : 'http://localhost:3001'
      }
    >
      <App />
    </Auth0Provider>
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
