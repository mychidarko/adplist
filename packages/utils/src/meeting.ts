import { hermes } from "./hermes";

export const JoinMeeting = async ({ apiUrl, organization, room, user, apiKey }: any) => {
  return hermes({
    authorization: apiKey,
    url: `${apiUrl}/organizations/${organization}/meetings/${room}/participant`,
    method: 'POST',
    data: {
      clientSpecificId: user.id,
      userDetails: {
        name: user.name,
        picture: user.picture,
      },
      roleName: user.role,
    },
  });
};
