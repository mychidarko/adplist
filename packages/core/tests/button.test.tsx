import { render } from '@testing-library/react';
import { Button } from '../src';

describe('<Component /> spec', () => {
  it('renders the component', () => {
    const h = render(<Button>Hello</Button>);
    expect(h.container.firstChild).toMatchSnapshot();
  });
});
