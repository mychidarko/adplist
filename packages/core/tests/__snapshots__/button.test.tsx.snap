// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> spec renders the component 1`] = `
<button
  class="c-btn btn btn-primary undefined"
  style="justify-content: center; text-align: center; pointer-events: all; align-items: center;"
>
  Hello
</button>
`;
